tests/lib/rules/no-unused-prop-types.js

```diff
@@ -1387,15 +1319,16 @@ ruleTester.run('no-unused-prop-types', rule, {
       code: [
         'const PagingBlock = function(props) {',
         '  return (',
-        '    <SomeChild {...props} />',
+        '    <span>{...props}</span>',
```

Comment:
as far as I can tell spread works with es6 but jsx spread inside a component does not. Simply removing babel and adding parserOptions fails with : `  message: 'Parsing error: Unexpected token ...',` 

I am happy to try any suggestions you might have.
```js 
  {
    code: [
      'const PagingBlock = function(props) {',
      '  return (',
      '    <span>{...props}</span>',
      ' );',
      '};',

      'PagingBlock.propTypes = {',
      '  nextPage: PropTypes.func.isRequired,',
      '  previousPage: PropTypes.func.isRequired,',
      '};'
    ].join('\n'),
    parserOptions: Object.assign({}, parserOptions, {ecmaVersion: 2017})
    }
```
error:
```js
    AssertionError: Should have no errors but had 1: [ { ruleId: null,
    fatal: true,
    severity: 2,
    source: '    <span>{...props}</span>',
    message: 'Parsing error: Unexpected token ...',
    line: 3,
    column: 12 } ]
      + expected - actual

      -1
      +0
      
      at testValidTemplate (node_modules/eslint/lib/testers/rule-tester.js:415:20)
      at Context.RuleTester.it (node_modules/eslint/lib/testers/rule-tester.js:545:25)

```